packages/esm-patient-chart-app/src/visit/hooks/useDeleteVisit.tsx

```diff
@@ -47,20 +52,23 @@ export function useDeleteVisit(visit: Visit, onVisitDelete = () => {}, onVisitRe
   const initiateDeletingVisit = () => {
     setIsDeletingVisit(true);
 
-    deleteVisit(visit?.uuid)
+    deleteVisit(activeVisit?.uuid)
       .then(() => {
-        // Update current visit data for critical components (useVisit hook)
-        mutateCurrentVisit();
+        // Update active visit data
+        mutateActiveVisit();
+        if (activeVisit.uuid == visitContext?.uuid) {
```

Comment:
```suggestion
        if (activeVisit.uuid == visitContext?.uuid && activeVisit.patient?.uuid === patientUuid) {
```

Might an additional patient UUID guard make this more robust?